src/components/Watch/FullScreen/FullScreen.tsx

```diff
@@ -0,0 +1,52 @@
+import "./FullScreen.css";
+import FullscreenIcon from "@mui/icons-material/Fullscreen";
+import FullscreenExitIcon from "@mui/icons-material/FullscreenExit";
+
+interface FullScreenProps {
+  videoRef: React.RefObject<HTMLVideoElement>;
+  isFullScreen: boolean;
+}
+
+function FullScreen({ videoRef, isFullScreen }: FullScreenProps) {
+  const handleFullScreenToggle = () => {
+    if (videoRef.current) {
+      if (!isFullScreen) {
+        // Enter full screen
+        if (videoRef.current.requestFullscreen) {
+          videoRef.current.requestFullscreen();
+        } else if (videoRef.current.mozRequestFullScreen) {
+          // Firefox
+          videoRef.current.mozRequestFullScreen();
+        } else if (videoRef.current.webkitRequestFullscreen) {
+          // Chrome, Safari and Opera
+          videoRef.current.webkitRequestFullscreen();
+        } else if (videoRef.current.msRequestFullscreen) {
+          // IE/Edge
+          videoRef.current.msRequestFullscreen();
+        }
+      } else {
+        // Exit full screen
+        if (document.exitFullscreen) {
+          document.exitFullscreen();
+        } else if (document.mozCancelFullScreen) {
+          // Firefox
+          document.mozCancelFullScreen();
+        } else if (document.webkitExitFullscreen) {
+          // Chrome, Safari and Opera
+          document.webkitExitFullscreen();
+        } else if (document.msExitFullscreen) {
+          // IE/Edge
+          document.msExitFullscreen();
+        }
+      }
```

Comment:
```suggestion
      } else {
        const document = window.document as Document & {
          mozCancelFullScreen: () => void;
          webkitExitFullscreen: () => void;
          msExitFullscreen: () => void;
        };
        
        // Exit full screen
        if (document.exitFullscreen) {
          document.exitFullscreen();
        } else if (document.mozCancelFullScreen) {
          // Firefox
          document.mozCancelFullScreen();
        } else if (document.webkitExitFullscreen) {
          // Chrome, Safari and Opera
          document.webkitExitFullscreen();
        } else if (document.msExitFullscreen) {
          // IE/Edge
          document.msExitFullscreen();
        }
      }
```

typescript로 인한 빌드 에러가 발생하는 두 번째 부분입니다. document 객체에 할당된 Document 내장 타입에도 exitFullScreen에 브라우저별 prefix가 붙은 메소드가 존재하지 않는 것으로 보입니다. 역시 Document에 새로운 타입이 추가된 

이처럼 typescript의 내장 타입을 강화한 타입들은 다른 페이지나 컴포넌트에서도 사용할 확률이 높으므로, 보통 src/types/utility.ts 같은 파일에 따로 빼서 공용으로 사용하는게 일반적입니다. 여유가 있으면 이렇게 따로 빼주세요!